reconcile/aws_version_sync/integration.py

```diff
@@ -67,7 +72,10 @@ class ExternalResource(BaseModel):
     resource_provider: str
     resource_identifier: str
     resource_engine: str
-    resource_engine_version: str
+    resource_engine_version: semver.VersionInfo | str
```

Comment:
We expect `resource_engine_version` to always be converted to `VersionInfo`, right?

can directly define

```python
    resource_engine_version: semver.VersionInfo
```

just need to use pre validator to convert it

```python
    @validator("resource_engine_version", pre=True)
```

Reply:
Strange, I had some mypy error w/o `str`. Nevertheless, now it's working [3c87702](https://github.com/app-sre/qontract-reconcile/pull/3940/commits/3c87702e2e8474947e132913363595cec090ba8c)

Reply:
🤦 fixed in f370bcb